token/delegation/delegation.go

```diff
@@ -142,8 +142,8 @@ func (t *Token) Nonce() []byte {
 }
 
 // Meta returns the Token's metadata.
-func (t *Token) Meta() *meta.Meta {
-	return t.meta
+func (t *Token) Meta() meta.ReadOnly {
```

Comment:
`meta` only needs to be mutable while applying `Option`s - can we "finalize" the immutable `meta` before the `Token` construction ends?

Reply:
Well actually no, that doesn't work. We need to pass the real Meta to `tokenPayloadModel`, so we'd have to make an accessor function, which would make that read-only much less read-only